plugins/vpp/l3plugin/vppcalls/dump_vppcalls.go

```diff
@@ -233,25 +245,146 @@ func (handler *arpVppHandler) DumpArpEntries() ([]*ArpEntry, error) {
 			return nil, err
 		}
 
-		var mac net.HardwareAddr = arpDetails.MacAddress
-		arp := &ArpEntry{
-			Interface:  arpDetails.SwIfIndex,
-			MacAddress: mac.String(),
-			Static:     uintToBool(arpDetails.IsStatic),
+		// ARP interface
+		ifName, _, exists := handler.ifIndexes.LookupName(arpDetails.SwIfIndex)
+		if !exists {
+			handler.log.Warnf("ARP dump: interface name not found for index %d", arpDetails.SwIfIndex)
 		}
-
-		var address net.IP
-		if arpDetails.IsIpv6 == 1 {
-			address = net.IP(arpDetails.IPAddress).To16()
+		// IP & MAC address
+		var ip, mac string
+		if uintToBool(arpDetails.IsIpv6) {
+			ip = fmt.Sprintf("%s", net.IP(arpDetails.IPAddress).To16().String())
 		} else {
-			address = net.IP(arpDetails.IPAddress[:4]).To4()
+			ip = fmt.Sprintf("%s", net.IP(arpDetails.IPAddress[:4]).To4().String())
+		}
+		mac = net.HardwareAddr(arpDetails.MacAddress).String()
+
+		// ARP entry
+		arp := &l3.ArpTable_ArpEntry{
+			Interface:   ifName,
+			IpAddress:   ip,
+			PhysAddress: mac,
+			Static:      uintToBool(arpDetails.IsStatic),
+		}
+		// ARP meta
+		meta := &ArpMeta{
+			SwIfIndex: arpDetails.SwIfIndex,
+		}
+
+		entries = append(entries, &ArpDetails{
+			Arp:  arp,
+			Meta: meta,
+		})
+	}
+
+	return entries, nil
+}
+
+// ProxyArpDetails holds info about proxy ARP interfaces/ranges as a proto modeled data
+type ProxyArpDetails struct {
```

Comment:
There should be details structure for proxy arp ranges and proxy arp interfaces as separate.